lib/SPIRV/libSPIRV/spirv.hpp

```diff
@@ -91,46 +91,51 @@ enum MemoryModel {
-    ExecutionModeVertexOrderCcw = 5,
-    ExecutionModePixelCenterInteger = 6,
-    ExecutionModeOriginUpperLeft = 7,
-    ExecutionModeOriginLowerLeft = 8,
-    ExecutionModeEarlyFragmentTests = 9,
-    ExecutionModePointMode = 10,
-    ExecutionModeXfb = 11,
-    ExecutionModeDepthReplacing = 12,
-    ExecutionModeDepthGreater = 14,
-    ExecutionModeDepthLess = 15,
-    ExecutionModeDepthUnchanged = 16,
-    ExecutionModeLocalSize = 17,
-    ExecutionModeLocalSizeHint = 18,
-    ExecutionModeInputPoints = 19,
-    ExecutionModeInputLines = 20,
-    ExecutionModeInputLinesAdjacency = 21,
-    ExecutionModeTriangles = 22,
-    ExecutionModeInputTrianglesAdjacency = 23,
-    ExecutionModeQuads = 24,
-    ExecutionModeIsolines = 25,
-    ExecutionModeOutputVertices = 26,
-    ExecutionModeOutputPoints = 27,
-    ExecutionModeOutputLineStrip = 28,
-    ExecutionModeOutputTriangleStrip = 29,
-    ExecutionModeVecTypeHint = 30,
-    ExecutionModeContractionOff = 31,
-    ExecutionModeInitializer = 33,
-    ExecutionModeFinalizer = 34,
-    ExecutionModeSubgroupSize = 35,
-    ExecutionModeSubgroupsPerWorkgroup = 36,
-    ExecutionModeMaxWorkgroupSizeINTEL = 5893,
-    ExecutionModeMaxWorkDimINTEL = 5894,
-    ExecutionModeNoGlobalOffsetINTEL = 5895,
-    ExecutionModeNumSIMDWorkitemsINTEL = 5896,
-    ExecutionModeMax = 0x7fffffff,
+  ExecutionModeInvocations = 0,
+  ExecutionModeSpacingEqual = 1,
+  ExecutionModeSpacingFractionalEven = 2,
+  ExecutionModeSpacingFractionalOdd = 3,
+  ExecutionModeVertexOrderCw = 4,
+  ExecutionModeVertexOrderCcw = 5,
+  ExecutionModePixelCenterInteger = 6,
+  ExecutionModeOriginUpperLeft = 7,
+  ExecutionModeOriginLowerLeft = 8,
+  ExecutionModeEarlyFragmentTests = 9,
+  ExecutionModePointMode = 10,
+  ExecutionModeXfb = 11,
+  ExecutionModeDepthReplacing = 12,
+  ExecutionModeDepthGreater = 14,
+  ExecutionModeDepthLess = 15,
+  ExecutionModeDepthUnchanged = 16,
+  ExecutionModeLocalSize = 17,
+  ExecutionModeLocalSizeHint = 18,
+  ExecutionModeInputPoints = 19,
+  ExecutionModeInputLines = 20,
+  ExecutionModeInputLinesAdjacency = 21,
+  ExecutionModeTriangles = 22,
+  ExecutionModeInputTrianglesAdjacency = 23,
+  ExecutionModeQuads = 24,
+  ExecutionModeIsolines = 25,
+  ExecutionModeOutputVertices = 26,
+  ExecutionModeOutputPoints = 27,
+  ExecutionModeOutputLineStrip = 28,
+  ExecutionModeOutputTriangleStrip = 29,
+  ExecutionModeVecTypeHint = 30,
+  ExecutionModeContractionOff = 31,
+  ExecutionModeInitializer = 33,
+  ExecutionModeFinalizer = 34,
+  ExecutionModeSubgroupSize = 35,
+  ExecutionModeSubgroupsPerWorkgroup = 36,
+  ExecutionModeMaxWorkgroupSizeINTEL = 5893,
+  ExecutionModeMaxWorkDimINTEL = 5894,
+  ExecutionModeNoGlobalOffsetINTEL = 5895,
+  ExecutionModeNumSIMDWorkitemsINTEL = 5896,
+  ExecutionModeDenormPreserve = 4459,
+  ExecutionModeDenormFlushToZero = 4460,
+  ExecutionModeSignedZeroInfNanPreserve = 4461,
+  ExecutionModeRoundingModeRTE = 4462,
+  ExecutionModeRoundingModeRTZ = 4463,
```

Comment:
Maybe better to move this a few lines up, so that the enum values are in their natural order.

Reply:
Fixed